### ISSUE TITLE
Rejected shares with GhostRider

### ISSUE THREAD

**Issue by Lonnegan** [xmrig/xmrig#2730]
Hi,

I've been mining Raptoreum with the cpu miner avx2 for weeks. In that time I had a few stale shares 0,2% but not a single rejected one.

Yesterday I switched from the cpu miner to xmrig 6.16 and despite of the short period of running there have already been 4 rejected shares. 4 is not much, indeed, but there must be something wrong. Rejected means, the share was not correct, defective, broken, somehow not right. Here I could find two:

[2021-11-27 14:38:04.649]  cpu      GhostRider algo 1: cn/fast (2 MB)
[2021-11-27 14:38:04.649]  cpu      GhostRider algo 2: cn/turtle (256 KB)
[2021-11-27 14:38:04.649]  cpu      GhostRider algo 3: cn/dark (512 KB)
[2021-11-27 14:38:25.354]  cpu      rejected (26/1) diff 55008 "high-hash" (44 ms)

[2021-11-27 16:17:36.169]  cpu      GhostRider algo 1: cn/dark-lite (256 KB)
[2021-11-27 16:17:36.169]  cpu      GhostRider algo 2: cn/lite (1 MB)
[2021-11-27 16:17:36.169]  cpu      GhostRider algo 3: cn/dark (512 KB)
[2021-11-27 16:17:37.542]  cpu      rejected (1713/5) diff 95559 "high-hash" (363 ms)

**Comment by SChernykh** [comment]
If you get a rejected share immediately after receiving new job from the pool, it's just an old share that was tested against a new job, i.e. stale share. Or, if you overclock your CPU and it was stable with cpuminer, it might not be stable with xmrig because the load on CPU is different.

**Comment by Lonnegan** [comment]
There are ones which are not immediately after a new job. I just chose those after algo change to help you finding a possible problem in on of the new cn-variants which were not part of xmrig up to 6.16.

The machines are not overclocked. These are fileservers with ECC ram running @stock 24/7.

**Comment by SChernykh** [comment]
Which pools was that?

**Comment by Lonnegan** [comment]
The same one as before, because I wanted to have the possibility to compare directly since that's not so simple with the permanently changing hashrate of GR: flockpool on the TLS port.

**Comment by SChernykh** [comment]
Can you check if you get these rejected shares soon after pool adjusts difficulty up or down?

**Comment by Lonnegan** [comment]
Well, I don't have logging enabled, I just can page up a bit, so I cannot see all errors anymore.

Regarding the rejeceted "stale-jobs" you are right. These occur milliseconds after the pool sent a new job and xmrig tried to return shares of the old one. Here for example:

[2021-11-27 17:26:07.128]  cpu      accepted (2488/30) diff 94104 (47 ms)
[2021-11-27 17:26:09.029]  net      new job from eu.flockpool.com:5555 diff
[2021-11-27 17:26:09.030]  cpu      rejected (2488/31) diff 94104 "stale-job
[2021-11-27 17:26:09.031]  cpu      GhostRider algo 1: cn/dark-lite (256 KB)
[2021-11-27 17:26:09.031]  cpu      GhostRider algo 2: cn/fast (2 MB)
[2021-11-27 17:26:09.031]  cpu      GhostRider algo 3: cn/turtle (256 KB)

You can see that the pool sent a new job at 17:26:09.029, so it rejects the simultaneously returned old one at 17:26:09.030 as "stale". I think that's normal.

The shares that pool rejected as "high-hash" I don't see noticeable timings. Here for example:

[2021-11-27 14:38:01.914]  cpu      accepted (26/0) diff 57457 (37 ms)
[2021-11-27 14:38:04.644]  net      new job from eu.flockpool.com:5555 diff 55008 algo ghostrider height 193486
[2021-11-27 14:38:04.649]  cpu      GhostRider algo 1: cn/fast (2 MB)
[2021-11-27 14:38:04.649]  cpu      GhostRider algo 2: cn/turtle (256 KB)
[2021-11-27 14:38:04.649]  cpu      GhostRider algo 3: cn/dark (512 KB)
[2021-11-27 14:38:25.354]  cpu      rejected (26/1) diff 55008 "high-hash" (44 ms)
[2021-11-27 14:38:39.514]  cpu      accepted (27/1) diff 55008 (39 ms)

The pool accepted shares of the old block height and difficulty 57457 at 14:38:01.914. Than the pool sent a new job with difficulty 55008 and new algo mix at 14:38:04.644. And than, at 14:38:25.354 the miner tried to return shares of that job, which you can recognize at the difficulty 55008 at 14:38:25.354, which the pool rejected not as stale but as "high-hash". That's 19 seconds (!) after the miner got the job. Further 6 seconds later at 14:38:39.514, xmrig returns further results of the same job and these were accepted.

**Comment by qanatozp** [comment]
Lonnegan, try to run it with rtm.suprnova.cc pool.
This pool spam with low difficulty task so you will fail shares more frequently.

**Comment by RussianE39** [comment]
I have exactly same results, 100% is rejected. In my case on ClearLinux, but when I tried precompiled static binary - everything works as expected. So I suspect its either gcc 11.2 bug or cmake configuration

**Comment by Lonnegan** [comment]
> Lonnegan, try to run it with rtm.suprnova.cc pool. This pool spam with low difficulty task so you will fail shares more frequently.

xmrig on suprnova:

[2021-11-28 01:50:31.511]  cpu      accepted (68/0) diff 31441 (92 ms)
[2021-11-28 01:50:38.893]  net      new job from rtm.suprnova.cc:4273 diff 31441 algo ghostrider height 193814
[2021-11-28 01:50:38.903]  cpu      GhostRider algo 1: cn/turtle-lite (128 KB)
[2021-11-28 01:50:38.903]  cpu      GhostRider algo 2: cn/turtle (256 KB)
[2021-11-28 01:50:38.905]  cpu      GhostRider algo 3: cn/dark-lite (256 KB)
[2021-11-28 01:50:38.962]  cpu      rejected (68/1) diff 31441 "job not found" (50 ms)

But a new reason, I've not seen on flockpool before: "job not found"

**Comment by Lonnegan** [comment]
> I have exactly same results

I don't think so! I do not have 100% rejected results. There are a few, meanwhile 12 within a day, that's around 0,2%. But before I had 0% rejected shares at flockpool, as I had before with xmrig and Haven. I'd mined Haven with xmrig for 15 month on Hashvault.pro and never had a single rejected share; a few stale shares of course, that's normal, but not a single invalid one.

**Comment by RussianE39** [comment]
From my experience, if you have only little bit of rejected shares, it could be RAM errors. Xmrig uses more memory than cpuminer-gr.

**Comment by Lonnegan** [comment]
I've never had rejected shares with xmrig in other, more traditional implementations like rx (Monero) or cn/xhv (Haven). GR support in xmrig is status alpha 0.1 ;-) So please don't be angry with me if I'm currently looking for bugs in xmrig rather than in pools or the hardware. Btw: my RAMs are not OC and are ECC, so RAM errors are currently the least likely source of errors.

**Comment by RussianE39** [comment]
You tried gcc built xmrig for windows?
